sklearn/metrics/pairwise.py

```diff
@@ -441,6 +441,45 @@ def pairwise_distances_argmin(X, Y, axis=1, metric="euclidean",
                                          batch_size=batch_size)[0]
 
 
+def haversine_distances(X, Y=None):
+    """ Compute the haversine distance between samples in X and Y
```

Comment:
Nit: maybe remove leading space and capitalize `haversine`

Reply:
Ready